test/DynamoCoreTests/DSCoreDataTests.cs

```diff
@@ -186,5 +191,359 @@ public void ParsingJSONInPythonReturnsSameResult()
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForNurbsCurveReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Curve_NurbsCurve_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("423356e2c8f84e00aa6c50e9bdb72c98", true);
+        }
+
+        [Test]
+        [Category("Failure")]
+        public void RoundTripForPolyCurveReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Curve_PolyCurve_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("423356e2c8f84e00aa6c50e9bdb72c98", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForPolygonReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Curve_Polygon_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("015f80f917374031b345b46b5a8d54ca", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForRectangleReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Curve_Rectangle_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("9754cbd66d4842419a6899f372a80aee", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForPointReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Point_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("71efc8c5c0c74189901707c30e6d5903", true);
+        }
+
+        [Test]
+        [Category("Failure")]
+        public void RoundTripForCylinderReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\Solid_Cylinder_JSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify objects match when serializing / de-serializing geometry type
+            AssertPreviewValue("07366adaf0954529b1ed39b240192c96", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void RoundTripForColorReturnsSameResult()
+        {
+            var color = DSCore.Color.ByARGB(25, 30, 35, 40);
```

Comment:
Is there any benefit to the preceding tests being implemented using graphs instead of implementing them to how these tests are written without involving graph deserialization? I find them easier to debug.

Reply:
Yeah... Sorry I was just following the pattern of the other tests for Json.  Also added benefit of much greater coverage.  We can create a follow up task and pull all the tests into unit test style if it is critical

Reply:
nah, not critical, maybe just preferred for future tests.